theano/tensor/elemwise.py

```diff
@@ -995,11 +1024,96 @@ def _c_all(self, node, nodename, inames, onames, sub):
             alloc_fortran = '0'
 
         alloc = ""
-        # We loop over the "real" outputs, i.e., those that are not
-        # inplace (must be allocated) and we declare/allocate/check
-        # them
-        for output, oname, odtype in izip(
-                real_outputs, real_onames, real_odtypes):
+
+        # Array referencing all inputs
+        ilength = len(_inames)
+        inArray = """
+        PyArrayObject **inRef[%(ilength)s];
+        """ % locals()
+        for iname, j in izip(_inames, range(len(_inames))):
+            inArray += """
+            inRef[%(j)i]=&%(iname)s;
+            """ % locals()
+
+        # Array referencing all outputs
+        olength = len(_onames)
+        outArray = """
+        PyArrayObject **outRef[%(olength)s];
+        """ % locals()
+        for oname, j in izip(_onames, range(len(onames))):
+            outArray += """
+            outRef[%(j)i]=&%(oname)s;
+            """ % locals()
+
+        alloc += inArray + outArray
+
+        # Declaration and initialization of the hardcoded olv_index
+        # used by the code generated in elemwise_cgen
+        alloc += """
+        int olv_index;
+        olv_index = %(j)i;
+        """ % locals()
+        if 'params' in sub:
+            # Get the destroy map from params
+            alloc += """
+               PyObject *tmpOut, *tmpIn;
+               Py_ssize_t pos;
+               pos = 0;
+               PyObject* destroyMap;
+               destroyMap =PyObject_GetAttrString(%(p)s,"inplace_pattern");
+               if( destroyMap == NULL){
+                  %(fail)s
+               }
+               """ % dict(p=sub['params'], fail=sub['fail'])
+            # Iterate over the python object to fill the destroy map arrays
+            # and make the output point to the corresponding input and
+            # decrease the reference of whatever the output contained
+            # prior to this
+
+            alloc += """
+               int lenDM;
+               lenDM = PyDict_Size(destroyMap);
```

Comment:
Hello! The problem seems to come from here. `inplace_pattern` (here the `destroyMap`) is a `frozendict`, not a dict (and `frozendict` class is not even a subclass of `dict`). I suggest you modify  `ElemwiseParams` to add a real dict field that you will then get in C code.
for example:
```python
class ElemwiseParams(object):
    def __init__(self, inplace_pattern):
        self._inplace_pattern = inplace_pattern  # frozendict is still useful for hashing method.
        self.inplace_pattern = self._inplace_pattern._dict  # Now `inplace_pattern` is a real dict.

    def __hash__(self):
        return hash(self._inplace_pattern)  # Using the frozendict for hashing.

```